README.md

```diff
@@ -109,6 +83,30 @@ c.Invoke(func(map map[string]int, slice []int, array [2]string) {
 })
 ```
 
+### Provide a Constructor
```

Comment:
constructor is the main use case and I'd love to keep it as one of the first sections